src/language_features/call_hierarchy.rs

```diff
@@ -32,56 +44,77 @@
     meta: EditorMeta,
     ctx: &mut Context,
     incoming_or_outgoing: bool,
-    result: Option<Vec<CallHierarchyItem>>,
+    results: Vec<(ServerName, Option<Vec<CallHierarchyItem>>)>,
 ) {
+    let result = results
+        .into_iter()
+        .find(|(_, response)| response.is_some())
+        .and_then(|(server_name, item)| Some((server_name, item.unwrap())));
+
     // TODO Can we get multiple items here?
-    let item = match result.and_then(|r| r.into_iter().next()) {
+    let (server_name, item) = match result
+        .and_then(|(server_name, r)| r.into_iter().next().and_then(|v| Some((server_name, v))))
+    {
         Some(item) => item,
         None => return,
     };
 
     if incoming_or_outgoing {
-        let params = CallHierarchyIncomingCallsParams {
-            item: item.clone(),
-            work_done_progress_params: WorkDoneProgressParams::default(),
-            partial_result_params: PartialResultParams::default(),
-        };
+        let mut params = HashMap::new();
+        params.insert(
+            server_name,
+            vec![CallHierarchyIncomingCallsParams {
+                item: item.clone(),
+                work_done_progress_params: WorkDoneProgressParams::default(),
+                partial_result_params: PartialResultParams::default(),
+            }],
+        );
 
         ctx.call::<CallHierarchyIncomingCalls, _>(
             meta,
-            params,
-            move |ctx: &mut Context, meta, result| {
-                format_call_hierarchy_calls(meta, ctx, incoming_or_outgoing, &item, &result);
+            RequestParams::Each(params),
+            move |ctx: &mut Context, meta, results| {
+                if let Some(result) = results.first() {
```

Comment:
ok so this shows the results of whichever server responds first.
This will probably need correction (what if one server always returns `[]`?).
It can be done in a follow-up. Maybe a priority, or maybe skip empty reslts..